Quadriga/src/main/java/edu/asu/spring/quadriga/rest/SearchNetworkRestController.java

```diff
@@ -108,10 +113,19 @@
             return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
         }
 
+        // Fetch ConceptPower entries related to the conceptId
+        ConceptpowerReply reply = conceptpowerConnector.getById(conceptId);
+        List<String> alternativeIdsForConcept = null;
+        if (reply != null && reply.getConceptEntry().size() > 0) {
+            alternativeIdsForConcept = reply.getConceptEntry().get(0).getAlternativeIdList();
+        }
+        if(alternativeIdsForConcept == null){
+            alternativeIdsForConcept = new ArrayList<String>();
+        }  
```

Comment:
If it means abstracting away the "business-logic" then it makes sense to write the logic in the service layer.  Also as per MVC, the controller is responsible only for directing requests and returning the response.
There is a trade-off between abstraction and dealing with dependencies. Bit confused about the design approach to take in general. Will shift the logic to the Service layer.